libDCM/estLocation.c

```diff
@@ -32,7 +33,16 @@ static void location_plane(int32_t* location)
 {
 	location[1] = ((lat_gps.WW - lat_origin.WW)/90); // in meters, range is about 20 miles
 	location[0] = long_scale((lon_gps.WW - lon_origin.WW)/90, cos_lat);
+//	location[2] = (alt_sl_gps.WW - alt_origin.WW)/100; // height in meters
+#ifdef USE_PRESSURE_ALT
+#warning "using pressure altitude instead of GPS altitude"
+	// division by 100 implies alt_origin is in centimeters; not documented elsewhere
+	// longword result = (longword/10 - longword)/100 : range
+	location[2] = ((get_barometer_altitude()/10) - alt_origin.WW)/100; // height in meters
+#else
 	location[2] = (alt_sl_gps.WW - alt_origin.WW)/100; // height in meters
+#endif // USE_PRESSURE_ALT
+
 }
 #else // !USE_EXTENDED_NAV

```

Comment:
What happens if the user is using the barometer without EXTENDED_NAVIGATION ? Is there any reason for keeping that option; if EXTENDED NAVIGATION works, then should we just move over to using it all the time ? (A question for Bill really).